src/hooks/useDatasetManager.js

```diff
@@ -144,7 +144,7 @@ export const useDatasetManager = () => {
 
     const processingDatasetId = isMyDatasetId(id)
       ? myDatasetId
-      : await createDataset() // creates new dataset ID for one-off download and shared dataset
+      : id || (await createDataset()) // creates a new dataset ID for one-off download or shared dataset
```

Comment:
```
    const processingDatasetId = isMyDatasetId(id)
      ? id
      : id || (await createDataset())

const processingDatasetId = id || (await createDataset())
```

Isn't this the same?

Reply:
Yes - cleaned up the check. The `dataset` prop is now passed to the form component, so there's no need to use the `MyDatasetId` value from localStorage here 👍

This PR is ready for another look. Thank you, David!